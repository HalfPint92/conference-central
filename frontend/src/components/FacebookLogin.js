import React from 'react';
import axios from 'axios';
import '../componentcss/LoginButtons.css';

const FacebookLogin = () => {
  // Function to handle Facebook login
  const handleFacebookLogin = async () => {
    try {
      // Send a GET request to the server to get the Facebook login URL.
      const { data } = await axios.get('http://localhost:5000/api/auth/facebook-login-url');

      // Redirect the user to the obtained Facebook login URL.
      // This will open the Facebook login page in a new tab or redirect the user to Facebook's login page.
      window.location.replace(data.url);
    } catch (error) {
      // If there is an error, log the error to the console.
      console.error(error);
    }
  };

  // The FacebookLogin component renders a button that triggers the Facebook login process when clicked.
  return (
    <button className="facebook-login-button" onClick={handleFacebookLogin}>
      Sign in with Facebook
    </button>
  );
};

export default FacebookLogin;
