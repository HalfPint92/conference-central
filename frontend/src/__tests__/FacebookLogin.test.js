import React from 'react';
import FacebookLogin from '../components/FacebookLogin';
import renderer from 'react-test-renderer';


test('renders correctly', () => {
    const tree = renderer.create(<FacebookLogin />).toJSON();
    expect(tree).toMatchSnapshot();
  });