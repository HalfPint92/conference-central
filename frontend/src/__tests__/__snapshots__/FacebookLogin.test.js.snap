// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="facebook-login-button"
  onClick={[Function]}
>
  Sign in with Facebook
</button>
`;
